File: DuAn_TN-FE/parse_eslint.js
const fs = require('fs');

const data = JSON.parse(fs.readFileSync('eslint_report_clean.json', 'utf8'));

let totalWarnings = 0;
const fileWarnings = [];

data.forEach(file => {
    if (file.warningCount > 0) {
        totalWarnings += file.warningCount;
        const warnings = file.messages
            .filter(msg => msg.severity === 1)
            .map(msg => ({
                ruleId: msg.ruleId,
                line: msg.line,
                message: msg.message
            }));
        fileWarnings.push({
            filePath: file.filePath,
            warningCount: file.warningCount,
            warnings: warnings
        });
    }
});

console.log(`Total Warnings: ${totalWarnings}`);
fileWarnings.forEach(fw => {
    console.log(`\nFile: ${fw.filePath}`);
    console.log(`Count: ${fw.warningCount}`);
    fw.warnings.forEach(w => {
        console.log(`  Line ${w.line}: [${w.ruleId}] ${w.message}`);
    });
});

File: DuAn_TN-FE/DuAn_TN-FE/parse_eslint.py
import json

with open('eslint_report.json', 'r', encoding='utf-8') as f:
    data = json.load(f)

total_warnings = 0
file_warnings = []

for file in data:
    if file['warningCount'] > 0:
        total_warnings += file['warningCount']
        warnings = []
        for msg in file['messages']:
            if msg['severity'] == 1: # 1 is warning
                warnings.append({
                    'ruleId': msg.get('ruleId'),
                    'line': msg.get('line'),
                    'message': msg.get('message')
                })
        file_warnings.append({
            'filePath': file['filePath'],
            'warningCount': file['warningCount'],
            'warnings': warnings
        })

print(f"Total Warnings: {total_warnings}")
for fw in file_warnings:
    print(f"\nFile: {fw['filePath']}")
    print(f"Count: {fw['warningCount']}")
    for w in fw['warnings']:
        print(f"  Line {w['line']}: [{w['ruleId']}] {w['message']}")
